src/electrum/server.rs

```diff
@@ -554,6 +543,48 @@ impl Connection {
         }
     }
 
+    #[inline]
+    fn handle_line(&mut self, line: &String) -> Value {
+        if let Ok(json_value) = from_str(line) {
+            match json_value {
+                Value::Array(mut arr) => {
+                    for cmd in &mut arr {
+                        // Replace each cmd with its response in-memory.
+                        *cmd = self.handle_value(cmd);
+                    }
+                    Value::Array(arr)
+                }
+                cmd => self.handle_value(&cmd),
+            }
+        } else {
+            // serde_json was unable to parse
+            invalid_json_rpc(line)
```

Comment:
I wonder if this is a good idea.

Essentially this responds to anything that's not valid JSON with a JSON error response instead of the current behavior of disconnecting the peer.

It might keep peers that are sending garbage or don't realize we're not speaking the same protocol.